### PR TITLE
initial upload of aws-ecr orb

### PR DESCRIPTION
**Pull request by ttrahan**
Note that this orb uses 'machine' executor. Please advise if an alternate executor is a better option.

### CODE REVIEW

File: src/aws-ecr/orb.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                        region: us-west-2
+                        account-url: myAwsAccountNum.dkr.ecr.myRegion.amazonaws.com 
+                        repo: myRepo
+                        tag: latest "
+orbs:
+  aws-cli: circleci/aws-cli@volatile
+
+executors:
+  default:
+    machine:
```
Comment (ndintenfass) [inline src/aws-ecr/orb.yml]: I suspect we should use remote docker inside some basic docker image (TODO: we don't seem to have a "generic" convenience image, which would be good for this purpose).

File: src/aws-ecr/orb.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
+version: 2.1
+
+description: "An orb for working with Amazon Elastic Container Registry (ECR)
```
[the remark this answers is not in the document]
Comment (ttrahan) [reply src/aws-ecr/orb.yml]: Corrected.

File: src/aws-ecr/orb.yml (hunk 1 above)
Comment (ttrahan) [reply src/aws-ecr/orb.yml]: I'll test this out with setup_remote_docker and commit a new executor. Hold for that.

File: src/aws-ecr/orb.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                build_and_push_image:
+                  jobs:
+                    - aws-ecr/build_and_push_image:
+                        context: myContext
+                        region: us-west-2
+                        account-url: myAwsAccountNum.dkr.ecr.myRegion.amazonaws.com 
+                        repo: myRepo
+                        tag: latest "
+orbs:
+  aws-cli: circleci/aws-cli@volatile
```
[the remark this answers is not in the document]
Comment (ttrahan) [reply src/aws-ecr/orb.yml]: I don't believe any version of aws-cli is currently published, though I've published and tested it with this orb in my personal namespace. I believe it's stable, though could use some clean up on the various descriptions.

File: src/aws-ecr/orb.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
+version: 2.1
+
+description: "An orb for working with Amazon Elastic Container Registry (ECR)
+              Sample usage within config.yml:
```
[the remark this answers is not in the document]
Comment (yundt) [reply src/aws-ecr/orb.yml]: @ttrahan I'll keep an eye on this PR; also, we should have a better idea when first-class usage examples go live tomorrow - will let you know!

**Comment by ttrahan** [comment]
@ndintenfass Machine executor replaced with setup_remote_docker. For the aws-cli orb, let me know if there's an action for me or if that will be handled by someone else.

File: src/aws-ecr/orb.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (yundt) [reply src/aws-ecr/orb.yml]: @ttrahan the first-class `examples` feature is in code review now - hopefully we can ship it sometime early next week.

**Comment by ttrahan** [comment]
@ndintenfass @yundt 
I've adde the new examples stanza added to orb definition.

File: src/aws-ecr/orb.yml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  necessary dependencies for executing the jobs and commands in this orb, 
+  except for the following required dependencies that must be configured in 
+  CircleCI in order to use this orb - Docker, environment variables for AWS 
+  login ($AWS_ACCESS_KEY_ID, $AWS_SECRET_ACCESS_KEY)
+
+  The source for this Orb can be found here:
+  https://github.com/CircleCI-Public/circleci-orbs/tree/master/src/aws-ecr
+
+examples:
+  Using job to build and push image to Amazon ECR:
```
Comment (yundt) [inline src/aws-ecr/orb.yml]: We opted for `examples` keys using the same format as e.g. workflow and job names, so only alphanumerical characters, `-`, and `_` are allowed. I suggest something like `simple_build_and_push`, and then incorporating `Using job to build and push image to Amazon ECR` into the `description` field.

The `examples` feature is already live, so you can validate your orb.yml with `circleci orb validate` CLI command.

**Comment by ttrahan** [comment]
Examples stanza updated and validated.

**Comment by ttrahan** [comment]
Note that an Orb for this should not be published until an Orb version is published for 'aws-cli', as it takes it as a dependency.

**Pull request merged**